### PR TITLE
Update zlib version

### PR DESCRIPTION
**Pull request by rjkroege**
I found that I needed the new version of zlib and to fetch it from https for the Docker to build.

### CODE REVIEW

**Comment by joseluisq** [comment]
Yep, every time there is a new release coming it supersedes the previous one like https://github.com/joseluisq/rust-linux-darwin-builder/pull/17. They do not preserve the tarballs history for some reason.

Probably we have to try to find out a second source.

**Comment by joseluisq** [comment]
Oh, there are https://www.zlib.net/fossils/

File: Dockerfile [hunk 1]
```diff
@@ -112,12 +112,12 @@ RUN set -eux \
...
 
 # zlib - http://zlib.net/
-ARG ZLIB_VERSION=1.2.13
+ARG ZLIB_VERSION=1.3
 
 RUN set -eux \
     && echo "Building zlib ${ZLIB_VERSION}..." \
     && cd /tmp \
-    && curl -LO "http://zlib.net/zlib-${ZLIB_VERSION}.tar.gz" \
+    && curl -LO "https://zlib.net/zlib-${ZLIB_VERSION}.tar.gz" \
```
Comment (joseluisq) [inline Dockerfile:120]: Let's use this one to be consistent from now on. 

```suggestion
    && curl -LO "https://www.zlib.net/fossils/zlib-${ZLIB_VERSION}.tar.gz" \
```

**Review by joseluisq — CHANGES_REQUESTED** [review]

**Comment by rjkroege** [comment]
I applied your suggestion.

**Review by joseluisq — APPROVED** [review]

**Pull request merged**